ci-operator/step-registry/sandboxed-containers-operator/trustee-install/sandboxed-containers-operator-trustee-install-commands.sh

```diff
@@ -0,0 +1,1387 @@
+
+# Configuration options
+# Set TRUSTEE_URL_USE_HTTP=true to use HTTP instead of HTTPS for Trustee URL (insecure - for testing only)
+TRUSTEE_URL_USE_HTTP="${TRUSTEE_URL_USE_HTTP:-false}"
+# Set TRUSTEE_URL_USE_NODEPORT=true to use nodeIP:nodePort instead of route hostname
+TRUSTEE_URL_USE_NODEPORT="${TRUSTEE_URL_USE_NODEPORT:-false}"
+# Set TRUSTEE_INSECURE_HTTP=true to enable insecure HTTP in KBS config (default: false)
+TRUSTEE_INSECURE_HTTP="${TRUSTEE_INSECURE_HTTP:-false}"
+# Set TRUSTEE_TESTING=true to use permissive resource policy for development/testing (default: false)
+TRUSTEE_TESTING="${TRUSTEE_TESTING:-false}"
+# Set TRUSTEE_ORG to customize the Organization (O) value in certificates (default: "Red Hat OpenShift")
+TRUSTEE_ORG="${TRUSTEE_ORG:-Red Hat OpenShift}"
+# Set TRUSTEE_CN to customize the Common Name (CN) value in certificates (default: "kbs-trustee-operator-system")
+TRUSTEE_CN="${TRUSTEE_CN:-kbs-trustee-operator-system}"
+# Set TRUSTEE_CATALOG_SOURCE_NAME to specify the catalog source for operator subscription
+# If not set, try to read from osc-config ConfigMap, otherwise default to "redhat-operators"
+if [ -z "${TRUSTEE_CATALOG_SOURCE_NAME:-}" ]; then
+    # Try to get catalog source name from osc-config ConfigMap created by env-cm step
+    OSC_CONFIG_CATALOG=$(oc get configmap osc-config -n default -o jsonpath='{.data.catalogsourcename}' 2>/dev/null || echo "")
+    if [ -n "$OSC_CONFIG_CATALOG" ]; then
+        TRUSTEE_CATALOG_SOURCE_NAME="$OSC_CONFIG_CATALOG"
+        echo "Using catalog source from osc-config: ${TRUSTEE_CATALOG_SOURCE_NAME}"
+    else
+        TRUSTEE_CATALOG_SOURCE_NAME="redhat-operators"
+        echo "Using default catalog source: ${TRUSTEE_CATALOG_SOURCE_NAME}"
+    fi
+else
+    echo "Using TRUSTEE_CATALOG_SOURCE_NAME environment variable: ${TRUSTEE_CATALOG_SOURCE_NAME}"
+fi
+
+# Function to wait for an operator subscription and CSV to finish installation
+# Parameters:
+#   $1: subscription_name - Name of the subscription
+#   $2: namespace - Namespace where the subscription exists
+#   $3: max_attempts - Maximum number of attempts (default: 60)
+#   $4: sleep_seconds - Seconds to sleep between attempts (default: 10)
+# Returns: 0 if successful, 1 if failed
+# Example:
+#   wait_for_operator_subscription "trustee-operator" "trustee-operator-system"
+#   wait_for_operator_subscription "my-operator" "my-namespace" 120 5
+wait_for_operator_subscription() {
+    local subscription_name="$1"
+    local namespace="$2"
+    local max_attempts="${3:-60}"
+    local sleep_seconds="${4:-10}"
+
+    if [ -z "$subscription_name" ] || [ -z "$namespace" ]; then
+        echo "Error: subscription_name and namespace are required parameters"
+        return 1
+    fi
+
+    # Wait for subscription to reach AtLatestKnown state
+    echo "Waiting for subscription '${subscription_name}' to be ready (state: AtLatestKnown)..."
+    local subscription_ready=false
+
+    for i in $(seq 1 "$max_attempts"); do
```

Comment:
I'd prefer looping over a specific deadline than number of iterations, it seems more predictable